Enhancement: Add Year to Job Run Timestamp in pipeline-stage-view-plugin

DeepData-Diver opened the pull request:
<!-- Please describe your pull request here. -->

### Testing done

<!-- Comment:
Provide a clear description of how this change was tested.
At minimum this should include proof that a computer has executed the changed lines.
Ideally this should include an automated test or an explanation as to why this change has no tests.
Note that automated test coverage is less than complete, so a successful PR build does not necessarily imply that a computer has executed the changed lines.
If automated test coverage does not exist for the lines you are changing, you must describe the scenario(s) in which you manually tested the change.
For frontend changes, include screenshots of the relevant page(s) before and after the change.
For refactoring and code cleanup changes, exercise the code before and after the change and verify the behavior remains the same.
-->

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md in your own repository 
-->

DeepData-Diver commented:
Addressed this Jenkins issue: https://issues.jenkins.io/browse/JENKINS-73507

jglick commented on pom.xml:
```diff
@@ -59,7 +59,7 @@
         <revision>2.35</revision>
         <gitHubRepo>jenkinsci/pipeline-stage-view-plugin</gitHubRepo>
         <bom>2.361.x</bom>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.462.1</jenkins.version>
```
Mismatch. See https://github.com/jenkinsci/archetypes/pull/737. Anyway this does not seem related to the rest of the PR and would better be reverted (there could be a separate PR to refresh dependencies).

DeepData-Diver replied:
Will revert tomorrow. Thank you

DeepData-Diver replied:
Done!

DeepData-Diver replied:
Done in the sense that the changes have been reverted back. Thanks

jglick commented on pom.xml.versionsBackup:
Delete.

DeepData-Diver replied:
Will delete tomorrow. Thank you

DeepData-Diver replied:
Done!

jglick commented on ui/src/main/js/view/templates/index.js:
```diff
@@ -100,6 +100,10 @@ registerHBSHelper('formatDate', function (date, toFormat) {
     if (toFormat == 'time') {
         return theDate.toLocaleTimeString(userLocale, {hour: '2-digit',minute: '2-digit', hour12: false });
     }
+    // New case for 'year'
```
```suggestion
```

jglick reviewed (changes requested):
Please be aware that this plugin is generally not maintained.

jglick commented:
I am not sure this is a net benefit; it breaks the layout a bit.

Chrome before

![Screenshot from 2024-09-16 12-22-16](https://github.com/user-attachments/assets/87a3d07a-b69d-452d-830d-0c8c6692dfa7)

and after

![Screenshot from 2024-09-16 12-23-01](https://github.com/user-attachments/assets/9aba4a62-809f-4b53-8f3d-dd97c0d85410)

and Firefox before

![Screenshot 2024-09-16 at 12-27-05 x Jenkins](https://github.com/user-attachments/assets/2c495bc4-4e5d-4975-9e60-1142c86a2908)

and after

![Screenshot 2024-09-16 at 12-27-13 x Jenkins](https://github.com/user-attachments/assets/b76213f1-0ec2-4249-967f-ac4e87e8f01a)

jglick commented:
Note that the issue suggests showing the year only if the build is more than a year old, which seems prudent. (Or a few months old at least. For that matter, maybe we should display the _day_ only if the build is more than a few hours old. Otherwise the time would suffice.)

DeepData-Diver commented:
Addressed PR comments. Please see new changes. Here is a sample image that illustrates changes:
![image](https://github.com/user-attachments/assets/6f258b5d-58a6-4d92-a9c8-31d077d304e8)

DeepData-Diver commented:
@jglick Please review PR. Thanks

DeepData-Diver commented:
Hello, just checking to see if more changes are required for this PR?

jglick commented:
Did not check; nobody maintains this plugin.

olamy commented:
let's get this merged

fahol-coop commented:
The last build broke the UI, as already mentioned before. The date field is now empty (blank space)
![image](https://github.com/user-attachments/assets/24c501d3-250c-47b8-8c10-3299b78695a0)

lppedd commented:
Same here. I'll revert to 2.34 in the meantime.

![image](https://github.com/user-attachments/assets/723a9ce2-f6ae-4326-b89e-4181f5976393)

lppedd commented:
Downgrading is trickier than what I expected. I can downgrade the Stage plugin, but not the REST one.

lppedd commented:
Folks, can we revert this? I can quickly fork and open a PR. This has caused more downsides than upsides.

olamy commented:
@lppedd sure please do it. Thanks

maichouni-mitek commented:
> Folks, can we revert this? I can quickly fork and open a PR. This has caused more downsides than upsides.

@lppedd , have you managed to downgrade? or is the PR still the route to go?
Thanks.
